packages/p2p/src/components/buy-sell/buy-sell-modal.jsx

```diff
@@ -161,97 +222,102 @@ const BuySellModal = ({ table_type, selected_ad, should_show_popup, setShouldSho
+                    className='buy-sell__modal'
+                    height_offset='80px'
+                    is_flex
+                    is_modal_open={should_show_popup}
+                    page_header_className='buy-sell__modal-header'
+                    page_header_text={generateModalTitle(formik_ref, my_profile_store, table_type, selected_ad)}
+                    pageHeaderReturnFn={onCancel}
+                    page_footer_parent={my_profile_store.should_show_add_payment_method_form ? '' : page_footer_parent}
+                    renderPageFooterChildren={() =>
+                        !my_profile_store.should_show_add_payment_method_form && (
                             <BuySellModalFooter
                                 is_submit_disabled={is_submit_disabled}
                                 onCancel={onCancel}
-                                onSubmit={submitForm.current}
+                                onSubmit={!has_rate_changed_recently ? submitForm.current : onSubmitWhenRateChanged}
                             />
-                        )}
-                    </Modal.Footer>
-                )}
-            </Modal>
-        );
-    }
-
-    return null;
+                        )
+                    }
+                    page_footer_className={
+                        my_profile_store.should_show_add_payment_method_form
+                            ? 'add-payment-method__footer'
+                            : 'buy-sell__modal-footer'
+                    }
+                >
+                    {table_type === buy_sell.SELL && is_account_balance_low && <LowBalanceMessage />}
+                    {!!error_message && buy_sell_store.form_error_code !== api_error_codes.MARKET_RATE_CHANGE && (
+                        <BuySellFormError />
+                    )}
+                    {my_profile_store.should_show_add_payment_method_form ? (
+                        <AddPaymentMethodForm formik_ref={formik_ref} should_show_separated_footer={true} />
+                    ) : (
+                        <Form
+                            advert={selected_ad}
+                            handleClose={onCancel}
+                            handleConfirm={onConfirmClick}
+                            setIsSubmitDisabled={setIsSubmitDisabled}
+                            setErrorMessage={setErrorMessage}
+                            setSubmitForm={setSubmitForm}
+                            setPageFooterParent={setPageFooterParent}
+                        />
+                    )}
+                </MobileFullPageModal>
+            </MobileWrapper>
+            <DesktopWrapper>
+                <Modal
+                    className='buy-sell__modal'
+                    height={table_type === buy_sell.BUY ? 'auto' : '649px'}
+                    width='456px'
+                    is_open={should_show_popup}
+                    title={generateModalTitle(formik_ref, my_profile_store, table_type, selected_ad)}
```

Comment:
Shouldn't this be: 

```suggestion
                    title={() => generateModalTitle(formik_ref, my_profile_store, table_type, selected_ad)}
```

Reply:
It has a type of `title: PropTypes.oneOfType([PropTypes.string, PropTypes.bool, PropTypes.node]),` though 👀